napari/utils/_proxies.py

```diff
@@ -40,24 +42,35 @@ class PublicOnlyProxy(wrapt.ObjectProxy, Generic[_T]):
 
     __wrapped__: _T
 
+    # __root limits the scope of the Proxy to types from the __root module
+    # set to empty string to recurse indefinitely
+    __root: str = __name__.split('.')[0]  # default to napari only
+
     def __getattr__(self, name: str):
-        if name.startswith('_'):
-            warnings.warn(
-                trans._(
-                    "Private attribute access in this context (e.g. inside a plugin widget or dock widget) is deprecated and will be unavailable in version 0.4.14",
-                    deferred=True,
-                ),
-                category=FutureWarning,
-                stacklevel=2,
-            )
-            # name = f'{type(self.__wrapped__).__name__}.{name}'
-            # raise AttributeError(
-            #     trans._(
-            #         "Private attribute access ('{name}') not allowed in this context.",
-            #         deferred=True,
-            #         name=name,
-            #     )
-            # )
+        if _SUNDER.match(name):
+            frame = sys._getframe(1) if hasattr(sys, "_getframe") else None
+            # allow napari to use private attributes
+            if frame and ROOT_DIR not in frame.f_back.f_code.co_filename:  # type: ignore
```

Comment:
If we are inside the napari code why you would like to add a performance penalty? In such a case object itself should be returned.

Reply:
> why you would like to add a performance penalty?

:joy:  

thanks...  i think I agree it should be fine to return directly in that case